WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -532,4 +488,56 @@ private static int GetInputTarget(int utxoCount, WasabiRandom rnd)
 
 		return targetInputCount;
 	}
+
+	private async Task<IEnumerable<TxOut>> ProgressOutputRegistrationPhaseAsync(uint256 roundId, ImmutableArray<AliceClient> registeredAliceClients, CancellationToken abortToken)
+	{
+		// Waiting for OutputRegistration phase, all the Alices confirmed their connections, so the list of the inputs will be complete.
+		var roundState = await RoundStatusUpdater.CreateRoundAwaiter(roundId, Phase.OutputRegistration, abortToken).ConfigureAwait(false);
+		var remainingTime = roundState.OutputRegistrationTimeout - RoundStatusUpdater.Period;
+		var outputRegistrationEndTime = DateTimeOffset.UtcNow + (remainingTime * 0.8);
+		var readyToSignEndTime = outputRegistrationEndTime + remainingTime * 0.2;
+
+		using CancellationTokenSource outputRegPhaseTimeoutCts = new(remainingTime + TimeSpan.FromMinutes(1));
+		using CancellationTokenSource combinedCts = CancellationTokenSource.CreateLinkedTokenSource(abortToken, outputRegPhaseTimeoutCts.Token);
+		var cancellationToken = combinedCts.Token;
```

Comment:
Would be good to move this closer to the place where it is used L525

Reply:
ACK